lib/streamlit/elements/widgets/button_group.py

```diff
@@ -0,0 +1,412 @@
+    current_form_id: str,
+    click_mode: ButtonGroupProto.ClickMode.ValueType,
+    selection_visualization: ButtonGroupProto.SelectionVisualization.ValueType = (
+        ButtonGroupProto.SelectionVisualization.ONLY_SELECTED
+    ),
+) -> ButtonGroupProto:
+    proto = ButtonGroupProto()
+
+    proto.id = widget_id
+    proto.default[:] = default_values
+    proto.form_id = current_form_id
+    proto.disabled = disabled
+    proto.click_mode = click_mode
+
+    for formatted_option in formatted_options:
+        proto.options.append(formatted_option)
+    proto.selection_visualization = selection_visualization
+    return proto
+
+
+class ButtonGroupMixin:
+    @gather_metrics("feedback")
+    def feedback(
+        self,
+        options: _FeedbackOptions = "thumbs",
+        *,
+        key: str | None = None,
+        disabled: bool = False,
+        on_change: WidgetCallback | None = None,
+        args: Any | None = None,
+        kwargs: Any | None = None,
+    ) -> int | None:
+        """Display a feedback widget.
+
+        This is useful to collect user feedback, especially in chat-based apps.
+
+        Parameters:
+        -----------
+        options: "thumbs", "faces", or "stars"
+            The feedback options displayed to the user. The options are:
+            - "thumbs" (default): displays a row of thumbs-up and thumbs-down buttons.
+            - "faces": displays a row of five buttons with facial expressions, each
+                depicting increasing satisfaction from left to right.
+            - "stars": displays a row of star icons typically used for ratings.
+        key : str or int
+            An optional string or integer to use as the unique key for the widget.
+            If this is omitted, a key will be generated for the widget
+            based on its content. Multiple widgets of the same type may
+            not share the same key.
+        disabled : bool
+            An optional boolean, which disables the multiselect widget if set
+            to True. The default is False. This argument can only be supplied
+            by keyword.
+        on_change : callable
+            An optional callback invoked when this multiselect's value changes.
+        args : tuple
+            An optional tuple of args to pass to the callback.
+        kwargs : dict
+            An optional dict of kwargs to pass to the callback.
+
+        Returns
+        -------
+        An integer indicating the user's selection, where 0 is the lowest
+        feedback and higher values indicate more positive feedback.
+        If no option was selected, returns None.
+            - For "thumbs": a return value of 0 is for thumbs-down and 1 for thumbs-up.
+            - For "faces" and "stars":
+                values range from 0 (least satisfied) to 4 (most satisfied).
+
+
+        Examples
+        --------
+        Example 1: Display a feedback widget with stars and show the selected sentiment
+        ```python
```

Comment:
nit: the examples should not be in markdown, but maybe something for Debbie to fix